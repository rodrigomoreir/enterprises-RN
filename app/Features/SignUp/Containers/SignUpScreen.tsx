import React from 'react';
import { useNavigation } from '@react-navigation/native';
import { 
  View, 
  Text, 
  Image, 
  KeyboardAvoidingView, 
  Platform, 
  TouchableOpacity, 
  ScrollView 
} from 'react-native';
import { Form } from '@unform/mobile'
import { getStyles } from './SignUpScreen.style'
import TextField from '../../../Components/TextField'
import { Button } from '../../../Components';
import images from '../../../Themes/Images'

const SignUpScreen: React.FC = () => {
  const navigation = useNavigation()
  const styles = getStyles()

  const navigateToSignUp = () => navigation.navigate('SignIn')

  return (
    <KeyboardAvoidingView
      style={{ flex: 1 }}
      behavior={Platform.OS === 'ios' ? 'padding' : undefined}
      enabled
    >
      <ScrollView
        keyboardShouldPersistTaps='handled'
        style={styles.containerScrollView}
        contentContainerStyle={styles.contentScrollView}
      >
        <View style={styles.container}>
          <View style={styles.imageContainer}>
            <Image source={images.logo} />
          </View>
          <Text style={styles.title}>Fa√ßa seu cadastro</Text>
          <View style={styles.inputsContainer}>       
          <Form onSubmit={() => {}}>   
            <TextField name={'name'} image={images.profile['20px']} placeholder={'Nome'} />
            <TextField name={'email'} image={images.profile['20px']} placeholder={'E-mail'} />
            <TextField 
              name={'senha'} 
              image={images.password['20px']} 
              placeholder={'Senha'} 
              secureTextEntry={true}
            />
          </Form>
          </View>
          <Button text={'Cadastrar'} onPress={() => {}}/>
          <TouchableOpacity style={styles.accessAccount} onPress={navigateToSignUp}>
            <Text style={styles.title}>Acessar conta</Text>
          </TouchableOpacity>
        </View>
      </ScrollView>
    </KeyboardAvoidingView>
  )
}

export default SignUpScreen;
