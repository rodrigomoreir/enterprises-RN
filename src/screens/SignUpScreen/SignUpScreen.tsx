import React from 'react'
import { useNavigation } from '@react-navigation/native'
import { Platform } from 'react-native'

import styled from 'styled-components/native'
import { screenWidth } from '../../utils'

import { Form } from '@unform/mobile'
import TextField from '../../core/TextField'
import { Button } from '../../core'
import images from '../../assets'

const SignUpScreen: React.FC = () => {
  const navigation = useNavigation()

  const navigateToSignUp = () => navigation.navigate('SignIn')

  return (
    <StyledKeyboardAvoidingView
      behavior={Platform.OS === 'ios' ? 'padding' : undefined}
      enabled>
      <StyledScrollView keyboardShouldPersistTaps="handled">
        <StyledViewContainer>
          <StyledViewImage>
            <StyledLogoImage source={images.logo} resizeMode={'contain'} />
          </StyledViewImage>
          <StyledTitle>Fa√ßa seu cadastro</StyledTitle>
          <StyledInputsContainer>
            <Form onSubmit={() => {}}>
              <TextField
                name={'name'}
                image={images.profile['20px']}
                placeholder={'Nome'}
              />
              <TextField
                name={'email'}
                image={images.profile['20px']}
                placeholder={'E-mail'}
              />
              <TextField
                name={'senha'}
                image={images.password['20px']}
                placeholder={'Senha'}
                secureTextEntry={true}
              />
            </Form>
          </StyledInputsContainer>
          <Button text={'Cadastrar'} onPress={() => {}} />
          <StyledSignUpButton onPress={navigateToSignUp}>
            <StyledTitle>Acessar conta</StyledTitle>
          </StyledSignUpButton>
        </StyledViewContainer>
      </StyledScrollView>
    </StyledKeyboardAvoidingView>
  )
}

const StyledKeyboardAvoidingView = styled.KeyboardAvoidingView`
  flex: 1;
`

const StyledScrollView = styled.ScrollView.attrs({
  contentContainerStyle: {
    flexGrow: 1,
    width: screenWidth,
    alignItems: 'center',
    justifyContent: 'space-between',
  },
})`
  width: ${screenWidth}px;
`
const StyledViewContainer = styled.View`
  flex: 1;
  align-items: center;
`
const StyledViewImage = styled.View`
  flex: 1;
  padding-top: 80px;
  width: ${screenWidth - 50}px;
`

const StyledLogoImage = styled.Image`
  width: 100%;
`

const StyledTitle = styled.Text`
  font-size: 20px;
  line-height: 26.38px;
  font-style: normal;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.darkGrey};
  align-self: center;
`
const StyledInputsContainer = styled.View`
  flex: 1;
  padding-vertical: 20px;
`
const StyledSignUpButton = styled.TouchableOpacity`
  margin-top: 25px;
  height: 60px;
  width: ${screenWidth}px;
  align-items: center;
  justify-content: center;
  border-top-width: 2px;
  border-color: ${({ theme }) => theme.colors.grey};
`

export default SignUpScreen
