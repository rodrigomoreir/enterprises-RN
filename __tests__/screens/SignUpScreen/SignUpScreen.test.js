import React from 'react'
import SignUpScreen from '../../../src/screens/SignUpScreen'

import {
  renderWithTheme,
  matchSnapshotWithTheme,
} from '../../../src/functions/testing'

describe('SignUp Screen', () => {
  test('matches snapshot', () => {
    matchSnapshotWithTheme(<SignUpScreen />)
  })

  test('renders correctly', () => {
    renderWithTheme(<SignUpScreen />)
  })
})
