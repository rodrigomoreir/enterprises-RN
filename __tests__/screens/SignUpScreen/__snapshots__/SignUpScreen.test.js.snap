// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Screen matches snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "space-between",
        "width": 750,
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Array [
        Object {
          "width": 750,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingTop": 80,
                "width": 700,
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../src/assets/logo_ioasys.png",
              }
            }
            style={
              Array [
                Object {
                  "width": "100%",
                },
              ]
            }
          />
        </View>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#696969",
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "lineHeight": 26.38,
              },
            ]
          }
        >
          Fa√ßa seu cadastro
        </Text>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingVertical": 20,
              },
            ]
          }
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderColor": "#696969",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "height": 50,
                    "marginVertical": 10,
                    "paddingHorizontal": 5,
                    "width": 670,
                  },
                  undefined,
                  false,
                  false,
                  ,
                ]
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/assets/usuario_20px.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "marginHorizontal": 5,
                    },
                  ]
                }
              />
              <TextInput
                allowFontScaling={true}
                defaultValue=""
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Nome"
                rejectResponderTermination={true}
                style={
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                  ]
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderColor": "#696969",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "height": 50,
                    "marginVertical": 10,
                    "paddingHorizontal": 5,
                    "width": 670,
                  },
                  undefined,
                  false,
                  false,
                  ,
                ]
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/assets/usuario_20px.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "marginHorizontal": 5,
                    },
                  ]
                }
              />
              <TextInput
                allowFontScaling={true}
                defaultValue=""
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="E-mail"
                rejectResponderTermination={true}
                style={
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                  ]
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderColor": "#696969",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "height": 50,
                    "marginVertical": 10,
                    "paddingHorizontal": 5,
                    "width": 670,
                  },
                  undefined,
                  false,
                  false,
                  ,
                ]
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/assets/senha.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "marginHorizontal": 5,
                    },
                  ]
                }
              />
              <TextInput
                allowFontScaling={true}
                defaultValue=""
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Senha"
                rejectResponderTermination={true}
                secureTextEntry={true}
                style={
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                  ]
                }
                underlineColorAndroid="transparent"
              />
            </View>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#87cefa",
              "borderRadius": 30,
              "height": 60,
              "justifyContent": "center",
              "opacity": 1,
              "width": 650,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontSize": 18,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Cadastrar
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#a9a9a9",
              "borderTopWidth": 2,
              "height": 60,
              "justifyContent": "center",
              "marginTop": 25,
              "opacity": 1,
              "width": 750,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#696969",
                  "fontSize": 20,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "lineHeight": 26.38,
                },
              ]
            }
          >
            Acessar conta
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
